src/Analyzer/ValidationUtils.cpp

```diff
@@ -26,6 +26,10 @@ namespace
 
 void validateFilter(const QueryTreeNodePtr & filter_node, std::string_view exception_place_message, const QueryTreeNodePtr & query_node)
 {
+    if (filter_node->getNodeType() == QueryTreeNodeType::LIST)
```

Comment:
Now error is better, I will adjust the test https://s3.amazonaws.com/clickhouse-test-reports/63405/c67a43e3a7354c3db488df44c05e5a0d56a454c5/fast_test.html

```
2024-05-07 16:51:58 Code: 36. DB::Exception: Received from localhost:9000. DB::Exception: Unsupported expression '*' in filter. (BAD_ARGUMENTS)
2024-05-07 16:51:58 (query: SELECT * WHERE *;  -- { serverError UNSUPPORTED_METHOD })
```